src/pages/MyKiva/MyKivaPageContent.vue

```diff
@@ -366,7 +366,17 @@ export default {
 			if (!preferences) return false;
 			const parsedPreferences = JSON.parse(preferences.preferences);
 			const existingGoals = parsedPreferences?.goals || [];
-			return existingGoals.length ? existingGoals[0] : null;
+			let goal = existingGoals.length ? existingGoals[0] : null;
+			// eslint-disable-next-line max-len
+			const achievementProgress = this.heroTieredAchievements.find(ach => ach.id === goal?.category)?.totalProgressToAchievement ?? 0;
+			if (goal) {
+				goal = {
+					...goal,
+					currentProgress: (achievementProgress - (goal?.currentLoanCount || 0)),
```

Comment:
I don't think this gives us the number we need. What we need to know is how many loans they've lent to since setting this goal.

So we probably need to set the loan count when the goal is set, then we can compare starting loan count against new loans, or something along those lines.

Another option is using the start date to compare against transactions, but if a user is really active, there might be too many transactions to pull without creating a special endpoint.

Reply:
Oh I see now, pulling from the category data. Yeah that works for the categories that match achievements. And yeah, I think maybe `loanTotalAtStart` might be clearer.